src/bluesearch/database/topic.py

```diff
@@ -295,3 +297,59 @@ def get_topics_for_pmc_article(
             journal_topics.append(descriptor["name"])
 
     return journal_topics
+
+
+def extract_info_from_zipfile(path: pathlib.Path) -> tuple[str, str]:
+    """Extract topic of a medRxiv/bioRxiv article.
+
+    Parameters
+    ----------
+    path
+        Path to a `.meca` file (which is nothing else
+        than a zip archive) with a fixed structured.
+
+    Returns
+    -------
+    topic : str
+        The subject area of the article.
+
+    journal : str
+        The journal the article was published in. Should be either
+        "medRxiv" or "bioRxiv".
+
+    Raises
+    ------
+    ValueError
+        Appropriate XML not found or the journal or topic are missing.
+    """
+    with zipfile.ZipFile(path) as myzip:
+        xml_files = [
+            x
+            for x in myzip.namelist()
+            if x.startswith("content/") and x.endswith(".xml")
+        ]
+
+        if len(xml_files) != 1:
+            raise ValueError(
+                "There needs to be exactly one .xml file inside of content/"
+            )
+
+        xml_file = xml_files[0]
+        pattern_topic = """<subj-group subj-group-type="hwp-journal-coll">\r\n<subject>(.*)</subject>"""  # noqa
```

Comment:
Ugly, too many characters. Feel free to suggest alternatives

Reply:
* Maybe using single quotes? `r'<subj-group subj-group-type="hwp-journal-coll">\r\n<subject>(.*)</subject>'`
* I'd use raw strings for patterns
* Is it necessary to write "\r\n"?
* Maybe it's safer to use `defusedxml` to parse this? Or is it too slow?

Reply:
* Raw strings - great point - it seems like it is the best practise
* "\r\n" - yeh, it is necessary since all the XML are using this carriage return + new line separator
* `defusedxml` - I was tying to avoid "parsing" the entire XML since in our pipeline it is done in a completely different step. I am not really that familiar with `defusedxml` and I am not sure if it can act "lazily". The fact that `re.search` only looks for the first occurrence was the main reason why I went for it.  However, great point! IMO if there are some issues with this `re` approach or it becomes unreadable we should stop using it.

Reply:
The single quotes and raw strings were done here: 1b99e9d78cd5470536db7057bd3a586b43536326

Reply:
I see the point about no parsing the whole XML. Main motivation for using a parser would be to not have to worry about equivalent syntax like `\n` vs `\r\n` and `"stuff"` vs `'stuff'` etc.

Reply:
Yeh, let's keep this one open and let's discuss it during the planning.

Reply:
### Discussed on 2022-01-11

- Note that we are already doing an `XML` parsing for `topic-extract` when the source is `PubMed` or `PMC` (see #525 and #533)
- Yes, there is some duplication of the parsing work, it is true
- Try to benchmark the runtime of extracting the topic from a pretty long `XML` file using the two approaches (`regex` vs. `XML` parsing) — if the parsing doesn't take too long, then I think we can go with that w/o worrying too much.

Reply:
See https://github.com/BlueBrain/Search/pull/535#issuecomment-1014457144 for the benchmark